tests/unit/package_managers/test_pip.py

```diff
@@ -2585,7 +2585,7 @@ def test_pip_requirement_copy(
         self,
         requirement_line: str,
         requirement_options: list[str],
-        new_values: dict[str, str],
+        new_values: Union[Dict[str, str], Dict[str, List[str]]],
```

Comment:
nitpick: `dict[str, Union[str, List[str]]]` is shorter to write.

Reply:
Mypy is once again acting up on dictionary unpacking on `new_requirement = original_requirement.copy(**new_values)`. It does work if the values are passed by keywords, but fair enough, I think the copy function needs a refactor anyway, so I'm okay with this as is.